pkg/commands/archive.go

```diff
@@ -137,3 +156,8 @@ func unTAR(baseDir string, r io.Reader) error {
 		}
 	}
 }
+
+func isSupportedArchive(source []byte) bool {
+	header, found := supportedArchiveHeaders[source[0]]
+	return found && bytes.Equal(source[:len(header)], header)
```

Comment:
In theory, if len(header) is larger than cap(souce), the code can throw. I guess buf.Peek probably always return a large slice so the code won't hit the issue. It's better to do length check before byte check.